setup-homebrew/main.sh

```diff
@@ -1,49 +1,48 @@
-#/bin/bash
+#!/bin/bash
 
 set -euo pipefail
 
-if which brew &>/dev/null; then
-    HOMEBREW_PREFIX="$(brew --prefix)"
-    HOMEBREW_REPOSITORY="$HOMEBREW_PREFIX/Homebrew"
-    HOMEBREW_CORE_REPOSITORY="$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-core"
-    brew update-reset "$HOMEBREW_REPOSITORY" "$HOMEBREW_CORE_REPOSITORY"
-else
-    HOMEBREW_PREFIX=/home/linuxbrew/.linuxbrew
-    HOMEBREW_REPOSITORY="$HOMEBREW_PREFIX/Homebrew"
-    sudo mkdir -p "$HOMEBREW_PREFIX"
-    sudo git clone --depth=1 https://github.com/Homebrew/brew "$HOMEBREW_REPOSITORY"
-
-    HOMEBREW_CORE_REPOSITORY="$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-core"
-    sudo mkdir -p "$HOMEBREW_CORE_REPOSITORY"
-    sudo rm -rf "$HOMEBREW_CORE_REPOSITORY"
-    sudo git clone --depth=1 https://github.com/Homebrew/linuxbrew-core "$HOMEBREW_CORE_REPOSITORY"
-
-    cd "$HOMEBREW_PREFIX"
-    sudo mkdir -p bin etc include lib opt sbin share var/homebrew/linked Cellar
-    sudo ln -sf ../Homebrew/bin/brew "$HOMEBREW_PREFIX/bin/"
-    cd -
-fi
+HOMEBREW_PREFIX="$(brew --prefix)"
+HOMEBREW_REPOSITORY="$(brew --repo)"
+HOMEBREW_CORE_REPOSITORY="$(brew --repo homebrew/core)"
+HOMEBREW_TAP_REPOSITORY="$(brew --repo "$GITHUB_REPOSITORY")"
 
-export PATH="$HOMEBREW_PREFIX/bin:$PATH"
 echo "::add-path::$HOMEBREW_PREFIX/bin"
 
+brew update-reset "$HOMEBREW_REPOSITORY" "$HOMEBREW_CORE_REPOSITORY"
+
 GEMS_PATH="$HOMEBREW_REPOSITORY/Library/Homebrew/vendor/bundle/ruby/"
+GEMS_HASH="$(shasum -a 256 "$HOMEBREW_REPOSITORY/Library/Homebrew/Gemfile.lock" | cut -f1 -d' ')"
+
 echo "::set-output name=gems-path::$GEMS_PATH"
-GEMS_HASH=$(shasum -a 256 "$HOMEBREW_REPOSITORY/Library/Homebrew/Gemfile.lock" | cut -f1 -d' ')
 echo "::set-output name=gems-hash::$GEMS_HASH"
 
-case "$GITHUB_REPOSITORY" in
-    */brew|*/*brew-core)
-        cd "$HOMEBREW_CORE_REPOSITORY"
-        rm -rf "$GITHUB_WORKSPACE"
-        ln -s "$HOMEBREW_CORE_REPOSITORY" "$GITHUB_WORKSPACE"
-        git fetch origin "$GITHUB_SHA"
-        git checkout --force -B master FETCH_HEAD
-        cd -
-    ;;
-esac
-
-if [ "$RUNNER_OS" = "Linux" ]; then
+# brew
+if [[ "$GITHUB_REPOSITORY" =~ ^.+/brew$ ]]; then
+    rm -rf "$GITHUB_WORKSPACE"
+    ln -s "$HOMEBREW_REPOSITORY" "$GITHUB_WORKSPACE"
```

Comment:
```suggestion
    NEW_GITHUB_WORKSPACE="$HOMEBREW_REPOSITORY"
```

Can do that in each of the `if`/`elif`s and then have a `if [[ -n "$NEW_GITHUB_WORKSPACE" ]]` which does all of this (shared) logic with `ln -s "$NEW_GITHUB_WORKSPACE" "$GITHUB_WORKSPACE"`

Reply:
I actually revisited it and those lines aren't always the same. I'd like to keep it the way it is now.

Reply:
Works for me 👍🏻